battles/helpers.py

```diff
@@ -75,22 +72,21 @@ def order_battle_pokemons(cleaned_data):
 def send_battle_result_email(battle):
     battle.refresh_from_db()
     trainers = [battle.trainer_creator, battle.trainer_opponent]
+    trainer_creator_team = TrainerTeam.objects.get(
```

Comment:
With the change suggested in the huge comment above, you could also access the teams here directly via the nested relations :)

Reply:
Done!